site/src/content/docs/docs/integrations/api/graphql.mdx

```diff
@@ -0,0 +1,234 @@
+  --format json > schema.json
+
+# using graphql-inspector
+npx @graphql-inspector/cli \
+  introspect https://graphql.union.build/v1/graphql \
+  --write schema.graphql
+```
+
+This schema can be used by API clients and editors like VSCode to provide a better developer experience.
+
+## Concepts
+
+The following definitions are used across the schema and documentation:
+
+- `source`: The chain or rollup where the packet is sent from, often prefixed such as `source_timestamp` or `source_transaction_hash`.
+- `destination`: The chain or roll-up that receives the packet. 
+- `sender`: the contract or [EOA](https://ethereum.org/en/developers/docs/accounts/) that made the transfer.
+- `receiver`: the contract or [EOA](https://ethereum.org/en/developers/docs/accounts/) that received the transfer.
+
+## Queries
+
+Below we have documented some common useful queries that are also leveraged in [app.union.build](https://app.union.build). For all queries, Union
+provides caches to increase performance. Use the syntax `@cached(ttl: ${time_in_seconds})` to cache the query. The cache key, which is used 
+to map the query to the response, is computed by hashing:
+
+- the GraphQL query.
+- the GraphQL operation name.
+- the GraphQL variables of the query.
+
+Caching can dramatically improve performance. Always add at least `@cached(ttl: 1)` to cache data between different windows, tabs, or users.
+
+### Transfers
+
+Transfers facilitated by Union (either `ucs01-relay-1` or `ics20-1`) are queryable using v1_transfers.
+
+```graphql
+query GetLatest10Transfers @cached(ttl: 1) {
+  v1_transfers(
+    limit: 10,
+    order_by: { source_timestamp: desc }
+  ) {
+    tokens {
+      denom
+      amount
+    }
+	  sender
+	  receiver
+  }
+}
+```
+
+It provides information on the sender, receiver, current stage of the transfer, and asset metadata.
+
+To query for specific addresses, add the `sender` and `receiver` to the where clause:
+
+```graphql
+query GetAllUserTransfers @cached(ttl: 1) {
+  v1_transfers(
+    order_by: { source_timestamp: desc }, 
+    where: {
+      _or: [
+        { sender: { _eq: "union1sw44hacrewvkuvm70zfxhwa9u6vm4f8yc8spqu" }},  
+        { receiver: { _eq: "union1sw44hacrewvkuvm70zfxhwa9u6vm4f8yc8spqu" }}
+      ]
+    }
+  ) {
+    tokens {
+      denom
+      amount
+    }
+    receiver
+    sender
+    source_chain { display_name }
+    destination_chain { display_name }
+  }
+}
+
+```
+Aggregate statistics about all transfers and packets are also available, such as the total amount of transfers and packets:
+
+```graphql
+query {
+ 	v1_statistics {
+    name
+    value
+  }
+}
+```
+
+### Packets
+
+Packets, used in general message passing, can be tracked using v1_packets, similar to v1_transfers.
+
+```graphql
+query GetLatest10Packets @cached(ttl: 1) {
+  v1_packets(
+    limit: 10,
+    order_by: { source_timestamp: desc }
+  ) {
+    source_chain { display_name }
+    destination_chain { display_name }
+		source_data
```

Comment:
renamed to `source_packet_hex` (because we also have `source_event_json`, which was originally named `source_json`)

Reply:
Is it always hex encoded though?

Reply:
ai, it's not. change to `source_packet_data`?

Reply:
yeah i think so